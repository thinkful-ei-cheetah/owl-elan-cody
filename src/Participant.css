ul {
    list-style-type: none;
    display: flex;
    flex-flow: column wrap;
}

ul li img {
    padding-right: 2em;
    max-width: 64px;
}

ul li {
    padding-top: 1em;
}

ul li:nth-child(1) {
    padding-top: 0;
}