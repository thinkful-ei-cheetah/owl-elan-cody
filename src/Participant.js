import React from 'react';
import './Participant.css'


function Participant({store}) {
    const participantsArray = store.map(item => 
    <li key = {item.id}>
        <img src={item.avatar} alt = "avatar" />
        {item.name}<br />
        {item.inSession ? 'in session ' : 'left session'}
    </li>)
    return (
        <ul>
            {participantsArray}
        </ul>
    );
}

export default Participant;