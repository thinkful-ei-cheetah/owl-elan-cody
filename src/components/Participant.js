import React from 'react';
import '../styles/Participant.css'


function Participant({store}) {
    const participantsArray = store.map(item => 
    <li key = {item.id}>
        <img src={item.avatar} alt = "avatar" />
        {item.name}<br />
        {item.inSession ? 'in session ' : 'left session'}
        {item.onStage ? 'on stage' : null}
    </li>)
    if (store.onStage && store.inSession) {
        store.onStage = !store.onStage;
        store.inSession = !store.inSession;
    }
    return (
        <div className="participants">
            {participantsArray}
        </div>    
    );
}

export default Participant;