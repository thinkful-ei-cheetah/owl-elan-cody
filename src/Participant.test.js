import React from 'react';
import ReactDOM from 'react-dom';
import Participant from './Participant';
import participants from './participants';
import renderer from 'react-test-renderer';

describe('Participant component', () => {
    it('renders without crashing' , () => {
        const div = document.createElement('div');
        ReactDOM.render(<Participant store={participants} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('renders UI as expected', () => {
        const tree = renderer
            .create(<Participant store={participants} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
});