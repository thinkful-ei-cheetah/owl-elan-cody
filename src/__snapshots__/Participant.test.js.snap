// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant component renders UI as expected 1`] = `
<ul>
  <li>
    <img
      alt="avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    Koren Templeton
    <br />
    in session 
  </li>
  <li>
    <img
      alt="avatar"
      src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
    />
    Caty Flucker
    <br />
    in session 
  </li>
  <li>
    <img
      alt="avatar"
      src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
    />
    Axe Kubicka
    <br />
    left session
  </li>
  <li>
    <img
      alt="avatar"
      src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
    />
    Frank Runciman
    <br />
    in session 
  </li>
  <li>
    <img
      alt="avatar"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
    Ashla Attwool
    <br />
    in session 
  </li>
</ul>
`;
